client/state/countries/reducer.js

```diff
@@ -1,25 +1,40 @@
 /** @format */
+
 /**
  * External dependencies
  */
-import { createReducer, combineReducers } from 'state/utils';
+import { combineReducers, createReducer } from 'state/utils';
 import {
+	COUNTRIES_DOMAINS_FETCH,
 	COUNTRIES_DOMAINS_UPDATED,
+	COUNTRIES_PAYMENTS_FETCH,
 	COUNTRIES_PAYMENTS_UPDATED,
+	COUNTRIES_SMS_FETCH,
 	COUNTRIES_SMS_UPDATED,
 } from 'state/action-types';
 
-const domains = createReducer( [], {
-	[ COUNTRIES_DOMAINS_UPDATED ]: ( countries, action ) => action.countries,
-} );
+const createListReducer = ( fetchActionType, updatedActionType ) => createReducer(
```

Comment:
Seems that can be more generic and go into `state/utils`:
```es6
const createListReducer = ( fetchActionType, updatedActionType, itemsPropName ) => { ... };
```
```es6
import { createListReducer } from 'state/utils';
const domains = createListReducer( COUNTRIES_DOMAINS_FETCH, COUNTRIES_DOMAINS_UPDATED, 'countries' );
```

Reply:
Looking at the content of `client/state/utils.js` I'm not sure this helper is generic enough to go there. It would live next to `createReducer()` which does other things than just structuring data.